sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AzureEventHubsKafkaOAuth2AutoConfiguration.java

```diff
@@ -0,0 +1,97 @@
+import org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.azure.spring.cloud.autoconfigure.context.AzureContextUtils.DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.KAFKA_OAUTH_CONFIGS;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.buildAzureProperties;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaAutoconfigurationUtils.needConfigureSaslOAuth;
+import static com.azure.spring.cloud.service.implementation.kafka.AzureKafkaPropertiesUtils.AZURE_TOKEN_CREDENTIAL;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for Azure Event Hubs Kafka support. Provide Managed Identity-based
+ * OAuth2 authentication for Event Hubs for Kafka on the basis of Spring Boot Autoconfiguration.
+ *
+ * @since 4.3.0
+ */
+@Configuration(proxyBeanMethods = false)
+@Import(AzureEventHubsKafkaOAuth2AutoConfiguration.AzureKafkaSpringCloudStreamConfiguration.class)
+@ConditionalOnClass(KafkaTemplate.class)
+public class AzureEventHubsKafkaOAuth2AutoConfiguration {
+
+    private final KafkaProperties kafkaProperties;
+    private final AzureTokenCredentialResolver tokenCredentialResolver;
+    private final AzureGlobalProperties azureGlobalProperties;
+    private final TokenCredential defaultTokenCredential;
+
+    AzureEventHubsKafkaOAuth2AutoConfiguration(KafkaProperties kafkaProperties,
+                                               AzureTokenCredentialResolver resolver,
+                                               @Qualifier(DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME) TokenCredential defaultTokenCredential,
+                                               AzureGlobalProperties azureGlobalProperties) {
+        this.kafkaProperties = kafkaProperties;
+        this.tokenCredentialResolver = resolver;
+        this.defaultTokenCredential = defaultTokenCredential;
+        this.azureGlobalProperties = azureGlobalProperties;
+    }
+
+    @Bean
+    DefaultKafkaConsumerFactoryCustomizer azureOAuth2KafkaConsumerFactoryCustomizer() {
+        Map<String, Object> updateConfigs = new HashMap<>();
+        Map<String, Object> consumerProperties = kafkaProperties.buildConsumerProperties();
+        if (needConfigureSaslOAuth(consumerProperties)) {
+            AzureKafkaProperties azureKafkaConsumerProperties = buildAzureProperties(consumerProperties,
+                azureGlobalProperties);
+            updateConfigs.put(AZURE_TOKEN_CREDENTIAL, buildCredentialFromProperties(azureKafkaConsumerProperties));
+            updateConfigs.putAll(KAFKA_OAUTH_CONFIGS);
+        }
```

Comment:
This seems like a duplicate of what we have in `azureOAuth2KafkaProducerFactoryCustomizer`?

Reply:
yeah, will extract to a method